types/datum.go

```diff
@@ -1281,11 +1281,11 @@ func (d *Datum) convertToMysqlDecimal(sc *stmtctx.StatementContext, target *Fiel
 		err = err1
 		dec.FromUint(val)
 	case KindMysqlJSON:
-		f, err1 := ConvertJSONToFloat(sc, d.GetMysqlJSON())
+		f, err1 := ConvertJSONToDecimal(sc, d.GetMysqlJSON())
```

Comment:
Please check L1624, and /types/convert.go L638

Reply:
thanks for notification, addressed, PTAL.